packages/contract-utils/upgradeable-macros/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+/// 1. Derives Upgradeable a) implements the interface; requires only the auth
+///    to be defined b) sets wasm version by taking the version from Cargo.toml
+///
+/// 2. Optionally derives Migratable when migration and rollback are defined.
+///
+///
+/// Example:
+/// ```rust,ignore
+/// #[derive(Upgradeable)]
+/// #[migratable]
```

Comment:
I couldn't see the code for the scenario where the user only provides `#[migratable]`.

I think there should be a `migratable` macro as well, and give an error, saying that `migratable` shouldn't be used without `upgradable`. 

And, `upgradable` can consume the macro `#[migratable]` (effectively, delete it), so that, `migratable` macro inner function won't be called when `#[derive(Upgradable)]` is present. 

I have another personal following this approach, can give examples if wanted.

Reply:
`#[migratable]` is an attribute of the derive macro and can't be used standalone https://github.com/OpenZeppelin/stellar-contracts/blob/e44e41dc578024454d715e27ac7732df7e8ad610/packages/contract-utils/upgradeable-macros/src/lib.rs#L40

Nevertherless, decided to follow your idea of having  `#[derive(Migratable)]`.